shell/platform/linux/fl_renderer.h

```diff
@@ -40,10 +40,38 @@ struct _FlRendererClass {
                            GdkScreen* screen,
                            EGLint visual_id);
 
-  // Virtual method called when Flutter needs a surface to render to.
-  EGLSurface (*create_surface)(FlRenderer* renderer,
-                               EGLDisplay display,
-                               EGLConfig config);
+  /**
+   * Virtual method called after a GDK window has been created.
+   * Does not need to be implemented.
+   */
+  void (*set_window)(FlRenderer* renderer, GdkWindow* window);
+
+  /**
+   * Virtual method to create a new EGL display.
+   */
+  EGLDisplay (*create_display)(FlRenderer* renderer);
+
+  /**
+   * Virtual method called when Flutter needs a surface to render to.
+   * @visible: (out): the visible surface that is created.
```

Comment:
Missing docstring for @display / @config. I know these weren't here before but if you add them might as well document all parameters.